operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/workflow/DefaultManagedWorkflow.java

```diff
@@ -78,14 +78,19 @@ public boolean isEmpty() {
   public Workflow<P> resolve(KubernetesClient client,
       ControllerConfiguration<P> configuration) {
     final var alreadyResolved = new HashMap<String, DependentResourceNode>(orderedSpecs.size());
+
+    // sharing the activation condition so no parallel requests are done for the same CRD
+    CRDPresentActivationCondition crdPresentActivationCondition =
```

Comment:
> imo and not recreated on each reconciliation (because then the cache is more or less useless since it gets nulled out on every reconciliation)

Not sure if following this part, it is not created on every reconciliation, this is called only once when the controller is initialized:
https://github.com/operator-framework/java-operator-sdk/blob/cdce182152d78b8ab3a97a45340255cd0545c3ab/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java#L96-L96

Or do you mean something else?

Reply:
> > imo and not recreated on each reconciliation (because then the cache is more or less useless since it gets nulled out on every reconciliation)
> 
> Not sure if following this part, it is not created on every reconciliation, this is called only once when the controller is initialized:
> 
> https://github.com/operator-framework/java-operator-sdk/blob/cdce182152d78b8ab3a97a45340255cd0545c3ab/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java#L96-L96
> 
> Or do you mean something else?

Ah yes, sorry, misread the diff… 🤦🏼